board/usb.h

```diff
@@ -643,29 +672,51 @@ void usb_irqhandler(void) {
       puts(" IN ENDPOINT\n");
     #endif
 
-    // this happens first
-    if (USBx_INEP(1)->DIEPINT & USB_OTG_DIEPINT_XFRC) {
-      #ifdef DEBUG_USB
-        puts("  IN PACKET SEND\n");
-      #endif
-      //USBx_DEVICE->DIEPEMPMSK = ~(1 << 1);
-    }
+    // Should likely check the EP of the IN request even if there is
+    // only one IN endpoint.
 
-    // *** IN token received when TxFIFO is empty
-    if (USBx_INEP(1)->DIEPINT & USB_OTG_DIEPMSK_ITTXFEMSK) {
-      #ifdef DEBUG_USB
+    // No need to set NAK in OTG_DIEPCTL0 when nothing to send,
+    // Appears USB core automatically sets NAK. WritePacket clears it.
+
+    // Handle the two interface alternate settings. Setting 0 is has
+    // EP1 as bulk. Setting 1 has EP1 as interrupt. The code to handle
+    // these two EP variations are very similar and can be
+    // restructured for smaller code footprint. Keeping split out for
+    // now for clarity.
+
+    //TODO add default case. Should it NAK?
+    switch(current_int0_alt_setting){
+    case 0: ////// Bulk config
```

Comment:
Agreed, the comment above explained my reasoning. I am happy to leave it as it is or compress it.